Sources/web3swift/Contract/EthereumContract.swift

```diff
@@ -6,137 +6,140 @@
 import Foundation
 import BigInt
 
-public struct EthereumContract: ContractProtocol {
+public class EthereumContract: ContractProtocol {
+
     public var transactionOptions: TransactionOptions? = TransactionOptions.defaultOptions
     public var address: EthereumAddress? = nil
 
-    var _abi: [ABI.Element]
+    public let abi: [ABI.Element]
```

Comment:
Just reminder that constant in reference type makes constant only the reference, not its values intead. So if it's required to deny user to modifying this filed we're should consider a private property with a public [computed] wrapper.

Reply:
I'm not sure I get your point.
The `EthereumContract` was changed to `class` from `struct`.
All variables except `address` are marked as `public let` and thus cannot be modified. 
All structs under `ABI.Element` are having their internal variables declared as `public let` (which is super good).

So there is no way for user of this class to modify something.